### ISSUE TITLE
can not enter guest os using kata-runtime exec

### ISSUE THREAD

**Issue by cheriL** [kata-containers/kata-containers#8447]
Hi community,
    I can not enter the VM if I use the alpine.img, `kata-runtime exec` has no response and no error return, but for ubuntu.img and centos-stream.img, it works fine. 
    All rootfs images are customized.
   **kata version:** **2.4.3**
   **hypervisor:** **qemu**
![image](https://github.com/kata-containers/kata-containers/assets/19817552/13e10ca1-3762-466e-9c7b-2cf883452994)

**Comment by cheriL** [comment]
```
11月 15 10:43:29 worker12 kata[15962]: time="2023-11-15T10:43:29.105621154+08:00" level=debug arch=amd64 command=exec default-kernel-parameters="systemd.unit=kata-containers.target systemd.mask=systemd-networkd.service systemd.mask=systemd-networkd.socket" name=kata-runtime pid=15962 source=katautils
11月 15 10:43:29 worker12 kata[15962]: time="2023-11-15T10:43:29.105663232+08:00" level=info arch=amd64 arguments="\"exec test\"" command=exec commit=42d18a88191721dea7a5f7b824b330730c516e98 name=kata-runtime pid=15962 source=runtime version=2.4.3
11月 15 10:43:29 worker12 kata[15828]: time="2023-11-15T10:43:29.110948902+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/test/console.sock name=containerd-shim-v2 pid=15828 sandbox=test source=virtcontainers subsystem=sandbox vmconsole="{\"msg\":\"get debug shell pid Pid(127)\",\"level\":\"INFO\",\"ts\":\"2023-11-15T02:43:29.100310594+00:00\",\"version\":\"0.1.0\",\"name\":\"kata-agent\",\"pid\":\"1\",\"subsystem\":\"debug-console-shell\",\"source\":\"agent\"}"
11月 15 10:43:29 worker12 kata[15828]: time="2023-11-15T10:43:29.111789614+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/test/console.sock name=containerd-shim-v2 pid=15828 sandbox=test source=virtcontainers subsystem=sandbox vmconsole="[    6.832504] sh[127]: segfault at 0 ip 000055e82f6c1c29 sp 00007ffe3a034b20 error 4 in busybox[55e82f6bf000+9c000]"
11月 15 10:43:29 worker12 kata[15828]: time="2023-11-15T10:43:29.112003198+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/test/console.sock name=containerd-shim-v2 pid=15828 sandbox=test source=virtcontainers subsystem=sandbox vmconsole="[    6.832668] Code: c9 09 00 e8 55 0b 00 00 6a 7f 5f e8 c1 e7 ff ff 53 48 8d 3d 88 c7 09 00 48 89 f3 e8 8d fb ff ff 48 8b 03 48 89 05 97 f8 0b 00 <80> 38 2d 75 0a 48 ff c0 48 89 05 88 f8 0b 00 48 8b 3d 81 f8 0b 00"
11月 15 10:43:29 worker12 kata[15828]: time="2023-11-15T10:43:29.112349342+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/test/console.sock name=containerd-shim-v2 pid=15828 sandbox=test source=virtcontainers subsystem=sandbox vmconsole="{\"msg\":\"master closed: Err(Os { code: 5, kind: Uncategorized, message: \\\"I/O error\\\" })\",\"level\":\"DEBG\",\"ts\":\"2023-11-15T02:43:29.101745266+00:00\",\"source\":\"agent\",\"name\":\"kata-agent\",\"version\":\"0.1.0\",\"subsystem\":\"debug-console-shell\",\"pid\":\"1\"}"
11月 15 10:43:29 worker12 kata[15828]: time="2023-11-15T10:43:29.11258723+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/test/console.sock name=containerd-shim-v2 pid=15828 sandbox=test source=virtcontainers subsystem=sandbox vmconsole="{\"msg\":\"debug console process exit code: Ok(Signaled(Pid(127), SIGSEGV, false))\",\"level\":\"INFO\",\"ts\":\"2023-11-15T02:43:29.102004951+00:00\",\"pid\":\"1\",\"subsystem\":\"debug-console-shell\",\"name\":\"kata-agent\",\"version\":\"0.1.0\",\"source\":\"agent\"}"
```

**Comment by skaegi** [comment]
Hi @cheriL -- this looks similar to a problem I encountered a few months ago. We added `bash` to our EXTRA_PKGS when we build the image and things work then.
```
EXTRA_PKGS="bash" AGENT_INIT=yes AGENT_VERSION=$kata_version ./rootfs.sh alpine 
```
I'm sorry didn't log the issue earlier! I didn't dig very deep but it looked to me like some issue in and around -- https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/console.rs#L36 -- that ends up requiring bash in the guest.
